src/audio/midi/MidiAudio.cpp

```diff
@@ -35,55 +32,29 @@ MidiAudio::MidiAudio(HWND windowHandle)
     }
 }
 
-MidiAudio::~MidiAudio() noexcept
-{
-    try
-    {
-#pragma warning(suppress : 26447)
-        // We acquire the lock here so the class isn't destroyed while in use.
-        // If this throws, we'll catch it, so the C26447 warning is bogus.
-        const auto lock = std::unique_lock{ _inUseMutex };
-    }
-    catch (...)
-    {
-        // If the lock fails, we'll just have to live with the consequences.
-    }
-}
-
-void MidiAudio::Initialize()
+void MidiAudio::BeginSkip() noexcept
 {
-    _shutdownFuture = _shutdownPromise.get_future();
+    _skip.SetEvent();
 }
 
-void MidiAudio::Shutdown()
+void MidiAudio::EndSkip() noexcept
 {
-    // Once the shutdown promise is set, any note that is playing will stop
-    // immediately, and the Unlock call will exit the thread ASAP.
-    _shutdownPromise.set_value();
+    _skip.ResetEvent();
 }
 
-void MidiAudio::Lock()
+void MidiAudio::PlayNote(HWND windowHandle, const int noteNumber, const int velocity, const std::chrono::milliseconds duration) noexcept
+try
 {
-    _inUseMutex.lock();
-}
+    if (_skip.is_signaled())
+    {
+        return;
+    }
 
-void MidiAudio::Unlock()
-{
-    // We need to check the shutdown status before releasing the mutex,
-    // because after that the class could be destroyed.
-    const auto shutdownStatus = _shutdownFuture.wait_for(0s);
-    _inUseMutex.unlock();
-    // If the wait didn't timeout, that means the shutdown promise was set,
-    // so we need to exit the thread ASAP by throwing an exception.
-    if (shutdownStatus != std::future_status::timeout)
+    if (_hwnd != windowHandle)
     {
-        throw Microsoft::Console::VirtualTerminal::StateMachine::ShutdownException{};

```

Comment:
Since we aren't using this exception anymore, we should get rid of the definition in [`statemachine.hpp`](https://github.com/microsoft/terminal/blob/f2b361c1463371ac2be681235470343ae48c23b5/src/terminal/parser/stateMachine.hpp#L71-L76), and the other bits of code associated with it that were added in PR #13208. There was a `catch` that was added to [`ControlCore::_connectionOutputHandler`](https://github.com/microsoft/terminal/blob/07201d6cd12870755efa7bb52e220b0b10102730/src/cascadia/TerminalControl/ControlCore.cpp#L1717-L1721) and another in  [`StateMachine::_SafeExecute`](https://github.com/microsoft/terminal/blob/f2b361c1463371ac2be681235470343ae48c23b5/src/terminal/parser/stateMachine.cpp#L2041-L2044). You may also need to add back a bunch of noexcepts in the `StateMachine` class if the audit complains.